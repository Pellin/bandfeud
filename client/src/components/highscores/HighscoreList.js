import React, { useState } from 'react';

import ShowBandlistModal from '../highscores/ShowBandlistModal';
import Crown from '../../icons/Crown';

const HighscoreList = ({ highscores }) => {
  const [showModal, setShowModal] = useState(false);
  const [bands, setBands] = useState([]);
  const [playerInfo, setPlayerInfo] = useState({
    index: 0,
    player: '',
    date: null,
    score: 0
  });
  const closeModal = () => {
    setShowModal(false);
    setBands([]);
  };
  const onHighscoreCLick = highscore => {
    setPlayerInfo({
      player: highscore.player,
      date: highscore.date,
      score: highscore.score
    });
    setBands(highscore.bands);
    setShowModal(true);
  };
  return (
    <>
      <div className="title-container">
        <Crown width={25} height="100%" className="crown-left" />
        <div className="highscore-title">HIGHSCORES</div>
        <Crown width={25} height="100%" className="crown-right" />
      </div>
      <table>
        <tbody className="highscore-body">
          {highscores.map(highscore => (
            <tr
              className="hs-row"
              onClick={() => onHighscoreCLick(highscore)}
              key={highscore._id}
            >
              <td className="hs-player">{highscore.player}</td>
              <td className="hs-score">{highscore.score}</td>
            </tr>
          ))}
        </tbody>
      </table>
      <ShowBandlistModal
        bands={bands}
        playerInfo={playerInfo}
        closeModal={closeModal}
        showModal={showModal}
      />
    </>
  );
};

export default HighscoreList;
