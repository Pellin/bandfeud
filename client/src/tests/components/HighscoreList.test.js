import React from 'react';
import { shallow } from 'enzyme';

import HighscoreList from '../../components/highscores/HighscoreList';

let highscores = [
  {
    date: 1567247152162.0,
    score: 97,
    player: 'falle'
  },
  {
    date: 1568387025126.0,
    score: 118,
    player: 'babs'
  },
  {
    date: 1568447073138.0,
    score: 123,
    player: 'mob pete'
  },
  {
    date: 1568734844461.0,
    score: 147,
    player: 'krille k'
  }
];

beforeEach(() => {});

it('should render HighscoreList correctly', () => {
  const wrapper = shallow(<HighscoreList highscores={highscores} />);

  expect(wrapper).toMatchSnapshot();
});
