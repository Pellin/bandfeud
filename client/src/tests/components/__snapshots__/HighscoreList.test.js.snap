// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render HighscoreList correctly 1`] = `
<Fragment>
  <div
    className="title-container"
  >
    <SvgCrown
      className="crown-left"
    />
    <div
      className="highscore-title"
    >
      HIGHSCORES
    </div>
    <SvgCrown
      className="crown-right"
    />
  </div>
  <table>
    <tbody
      className="highscore-body"
    >
      <tr
        className="hs-row"
        key="one"
        onClick={[Function]}
      >
        <td
          className="hs-player"
        >
          falle
        </td>
        <td
          className="hs-score"
        >
          97
        </td>
      </tr>
      <tr
        className="hs-row"
        key="two"
        onClick={[Function]}
      >
        <td
          className="hs-player"
        >
          babs
        </td>
        <td
          className="hs-score"
        >
          118
        </td>
      </tr>
      <tr
        className="hs-row"
        key="three"
        onClick={[Function]}
      >
        <td
          className="hs-player"
        >
          mob pete
        </td>
        <td
          className="hs-score"
        >
          123
        </td>
      </tr>
      <tr
        className="hs-row"
        key="four"
        onClick={[Function]}
      >
        <td
          className="hs-player"
        >
          krille k
        </td>
        <td
          className="hs-score"
        >
          147
        </td>
      </tr>
    </tbody>
  </table>
  <Link
    className="back-button"
    to="/"
  >
    BACK
  </Link>
  <ShowBandlistModal
    bands={Array []}
    closeModal={[Function]}
    playerInfo={
      Object {
        "date": null,
        "index": 0,
        "player": "",
        "score": 0,
      }
    }
    showModal={false}
  />
</Fragment>
`;
